src/main/java/com/fulfilment/application/monolith/warehouses/adapters/restapi/WarehouseResourceImpl.java

```diff
@@ -1,15 +1,21 @@
 package com.fulfilment.application.monolith.warehouses.adapters.restapi;
 
+import com.fulfilment.application.monolith.warehouses.adapters.database.WarehouseRepository;
 import com.warehouse.api.WarehouseResource;
 import com.warehouse.api.beans.Warehouse;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
 import jakarta.validation.constraints.NotNull;
 import java.util.List;
 
+@RequestScoped
 public class WarehouseResourceImpl implements WarehouseResource {
 
+  @Inject private WarehouseRepository warehouseRepository;
+
   @Override
   public List<Warehouse> listAllWarehousesUnits() {
-    return List.of();
+    return warehouseRepository.getAll().stream().map(this::toWarehouseResponse).toList();
```

Comment:
This is good to see if the candidates also take the hint of the style and try to work in the rest of the application with it.

Reply:
Brillant, I haven't thought about that and this is indeed a very good "induction" point. Nice.!